src/server/groups/read.ts

```diff
@@ -64,6 +64,56 @@ export async function readMembershipsOfUser(
     }))
 }
 
+export async function readGroupsOfUser(id: number) {
```

Comment:
I think it might be better to just include all the grouptypes from the start, it avoids running a lot of queries in parallell, and that is how it has been done in other places (visibility). Input from anyone else?